src/utils/marketsAndNetworksConfig.ts

```diff
@@ -143,6 +145,110 @@ export const isFeatureEnabled = {
   permissions: (data: MarketDataType) => data.enabledFeatures?.permissions,
 };
 
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+class StaticJsonRpcBatchProvider extends ethersProviders.JsonRpcBatchProvider {
+  async detectNetwork(): Promise<Network> {
+    let network = this.network;
+    if (network == null) {
+      network = await super.detectNetwork();
+
+      if (!network) {
+        logger.throwError('no network detected', Logger.errors.UNKNOWN_ERROR, {});
+      }
+
+      // If still not set, set it
+      if (this._network == null) {
+        // A static network does not support "any"
+        defineReadOnly(this, '_network', network);
+
+        this.emit('network', network, null);
+      }
+    }
+    return network;
+  }
+}
+
+/**
+ * The provider will rotate rpcs on error.
+ */
+class RotationProvider extends ethersProviders.BaseProvider {
+  readonly providers: StaticJsonRpcBatchProvider[];
+  private currentProviderIndex = 0;
+  private lastRotation = 0;
+  private retriedFirst = false;
+
+  constructor(urls: string[], chainId: number) {
+    super(chainId);
+    this.providers = urls.map((url) => new StaticJsonRpcBatchProvider(url, chainId));
+  }
+
+  /**
+   * Add a delay after completing a rotation, so we don't ddos a provider
+   */
```

Comment:
Can we update/add a comment that mimics what you have in the PR descriptoin for this?  i.e. that if all RPC providers fail we will delay 5000ms and try again, and that's what this is doing.   Also, I think we should store the value as a constant:

`const RETRY_DELAY = 5000`

Reply:
Added these as variables in the RotationProvider class

Reply:
Also updated the class description and added comments for individual lines in the rotation logic